mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -129,6 +136,22 @@ final class DecisionTreeRegressionModel private[ml] (
     rootNode.predictImpl(features).prediction
   }
 
+  /** We need to update this function if we ever add other impurity measures. */
+  protected def predictVariance(features: Vector): Double = {
+    rootNode.predictImpl(features).impurityStats.calculate()
+  }
+
+  override protected def transformImpl(dataset: DataFrame): DataFrame = {
```

Comment:
Good points! I have updated my code and add test for turning off ```predictionCol``` but turning on ```varianceCol```, it works as expected. 
I have noticed that it will output warning log for turning off ```predictionCol``` in [```PredictionModel```](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/Predictor.scala#L178), I think we should remove it because if users set ```predictionCol``` to ```""```(null string) means they don't want to output ```predictionCol``` in purpose and we should not notice them. However this issue is not relevant to this PR, I will file another JIRA to track it.